### DISCUSSION TITLE
how  use /auth/google/callback  for Android native app.

### DISCUSSION THREAD

**Discussion by fredycampino** [strapi/strapi#7444] [Legacy Topics]
After to do sign in with **GoogleSignInClient**  it has the id_token and/or  ServerAuthCode. But it does not provide the acces_token  that it needs for call  to teh callback  enpoint. **/auth/google/callback**. 

There are other endpoint to register the user? 
How can i custom the callback?

**Comment by derrickmehaffy** [comment]
@petersg83 @alexandrebodin do you have any suggestions on this topic? I don't think there should be a difference between desktop/mobile response from the google API here, is this an upstream API change that we weren't aware of?

**Comment by petersg83** [comment]
Hi,
Thanks for the question!
As of today Strapi doesn't natively support the mobile workflow. We use grant and it only support the web worflow: https://github.com/simov/grant/issues/104.
You will need to customise this authentication on your side by creating some new controllers: https://strapi.io/documentation/v3.x/concepts/controllers.html#custom-controllers

**Reply by fredycampino** [reply]
ok, Thanks for you replpy. one think more @petersg83 , where can i see the controller code for the current flow fot /auth/google/callback.

**Reply by petersg83** [reply]
yes, it's here: https://github.com/strapi/strapi/blob/master/packages/strapi-plugin-users-permissions/controllers/Auth.js#L21 :)
